docs/reference/openapi-rules.md

```diff
@@ -1,3 +1,544 @@
-## OpenAPI Rules
+# OpenAPI Rules
 
-Coming soon...
\ No newline at end of file
+Spectral has three rulesets built in: 
+
+- [oas](#oas)
+- [oas2](#oas2)
+- [oas3](#oas3)
+
+_OAS is shorthand for OpenAPI Specification._
+
+In your ruleset file you can add `extends: "spectral:oas"` and you'll get everything from `spectral:oas2` and `spectral:oas3`, meaning you can lint either type of document and only the relevant rules will apply thanks to [formats](../getting-started/rulesets.md#formats).
+
+Let's look at the rules in these rulesets.
+
+## oas
+
+### operation-2xx-response
+
+Operation must have at least one `2xx` response. Any API operation (endpoint) can fail but presumably it is also meant to do something constructive at some point. If you forget to write out a success case for this API, then this rule will let you know.
+
+**Recommended:** Yes
+
+**Good Example**
+
+```yaml
+paths:
+  /path:
+    get:
+      responses:
+        418:
+          description: teapot
+```
+
+### operation-operationId-unique
+
+Every operation must have a unique `operationId`. 
+
+Why? A lot of documentation systems use this as an identifier, some SDK generators convert them to a method name, all sorts of things like that. Giving each one a unique name avoids all sorts of problems.
+
+**Recommended:** Yes
+
+**Bad Example**
+
+```yaml
+paths:
+  /pet:
+    patch:
+      operationId: "update-pet"
+      responses: 
+        200:
+          description: ok
+    put:
+      operationId: "update-pet"
+      responses: 
+        200:
+          description: ok
+```
+
+**Good Example**
+
+```yaml
+paths:
+  /pet:
+    patch:
+      operationId: "update-pet"
+      responses: 
+        200:
+          description: ok
+    put:
+      operationId: "replace-pet"
+      responses: 
+        200:
+          description: ok
+```
+
+### operation-parameters
+
+Operation parameters are unique and non-repeating.
+
+1. Operations must have unique `name` + `in` parameters.
+2. Operation cannot have both `in: body` and `in: formData` parameters. (OpenAPI v2.0)
+3. Operation must have only one `in: body` parameter. (OpenAPI v2.0)
+
+**Recommended:** Yes
+
+### path-params
+
+Path parameters are correct and valid.
+
+1. for every param referenced in the path string (i.e: `/users/{userId}`), the parameter must be defined in either
+   `path.parameters`, or `operation.parameters` objects.
+
+2. every `path.parameters` and `operation.parameters` parameter must be used in the path string
+
+**Recommended:** Yes
+
+### contact-properties
+
+The `info-contact` rule will ask you to put in a contact object, and this rule will make sure its full of the most useful properties: `name`, `url` and `email`. 
```

Comment:
Maybe add a link to the `info-contact` rule?

Or, even better, from the reader standpoint, document the `info-contact` rule just before the `contact-properties` rule (However, topological sort OCD might strongly disagree with that approach ;-) )